Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -323,6 +347,99 @@ - (void) _connect {
     }
 }
 
+- (BOOL) connectToHostWithName: (NSString*)hostname
+                          port: (NSInteger)port
+              networkInterface: (NSString* _Nullable)interface
+                   inputStream: (NSInputStream**)inputStream
+                  outputStream: (NSOutputStream**)outputStream
+                         error: (NSError** _Nullable)error
+{
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_socktype = SOCK_STREAM;
+    
+    struct addrinfo* addr;
+    const char* cHost = [hostname cStringUsingEncoding: NSUTF8StringEncoding];
+    const char* cPort = [[NSString stringWithFormat: @"%ld", (long)port]
+                         cStringUsingEncoding: NSUTF8StringEncoding];
+    if (getaddrinfo(cHost, cPort, &hints, &addr)) {
+        NSString* m = [NSString stringWithFormat: @"failed to get address info, %s", strerror(errno)];
+        return [self populateError: error errNo: errno msg: m];
+    }
+    
+    int sockfd = socket(addr->ai_family, addr->ai_socktype, addr->ai_protocol);
+    if (sockfd < 0) {
+        freeaddrinfo(addr);
+        
+        NSString* m = [NSString stringWithFormat: @"failed to create socket, %s", strerror(errno)];
```

Comment:
I think we should include addr->ai_family, addr->ai_socktype, addr->ai_protocol in the log as well.